packages/contracts-bedrock/contracts/echidna/FuzzBurn.sol

```diff
@@ -0,0 +1,71 @@
+pragma solidity 0.8.15;
+
+import { Burn } from "../libraries/Burn.sol";
+import { StdUtils } from "forge-std/Test.sol";
+
+contract EchidnaFuzzBurnEth is StdUtils {
+    bool failedEthBurn;
+
+    /**
+     * @notice Takes an integer amount of eth to burn through the Burn library and
+     * updates the contract state if an incorrect amount of eth moved from the contract
+     */
+    function testBurn(uint256 _value) public {
+        // cache the contract's eth balance
+        uint256 preBurnBalance = address(this).balance;
+        uint256 value = bound(_value, 0, preBurnBalance);
+
+        // execute a burn of _value eth
+        Burn.eth(value);
+
+        // check that exactly value eth was transfered from the contract
+        unchecked {
+            if (address(this).balance != preBurnBalance - value) {
+                failedEthBurn = true;
+            }
+        }
+    }
+
+    function echidna_burn_eth() public view returns (bool) {
+        // ASSERTION: The amount burned should always match the amount passed exactly
+        return !failedEthBurn;
+    }
+}
+
+contract EchidnaFuzzBurnGas is StdUtils {
+    bool failedGasBurn;
+
+    /**
+     * @notice Takes an integer amount of gas to burn through the Burn library and
+     * updates the contract state if at least that amount of gas was not burned
+     * by the library
+     */
+    function testGas(uint256 _value) public {
+        // cap the value to the max resource limit
+        uint256 MAX_RESOURCE_LIMIT = 8_000_000;
```

Comment:
Implicit assumption here that the amount of gas available is at least 8m + some buffer to get to the end of execution safely.

Reply:
Well, I think the assumption is that it's at least `value` which is set below by modulating the argument into the range [0,MAX_RESOURCE_LIMIT].

Either way, if there's insufficient gas this whole thing will revert, which I think should be fine (echidna doesn't treat that as a failure).